src/test/java/org/jabref/model/groups/SearchGroupTest.java

```diff
@@ -26,6 +27,7 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
+@AllowedToUseLogic("because it needs access to lucene manager")
```

Comment:
Remember that everytime this annotation is used, this is a clear indication that there is something wrong with the architecture, but the effort is right now to large to fix it.
So always ask these questions before adding this annotation: Is the test you have made really a test for SearchGroup or is it a test for LuceneManager? Is the class SearchGroup also using LuceneManager, is it also annotated? Can the test be more detailed, more concentrated on things that don't need the LuceneManager?

Reply:
I’ve thought a lot about this, and SearchGroup was use LuceneManager and allowed to use logic, but I removed it from SearchGroups in this commit [`1005685` (#11542)](https://github.com/JabRef/jabref/pull/11542/commits/1005685d1946c845efef137b33780c5c5a730d32).

In testing SearchGroups, it turns out we are actually testing LuceneManager, since SearchGroups fully depend on the search results returned by LuceneManager. Before this PR, SearchGroup tests also checked the search grammar to see if entries were correctly matched and included in the groups.

Now we have separate tests for LuceneManager, do we need any additional tests for SearchGroups, or should we just remove the existing tests?

Reply:
Depends on the coverage. In case tests are removed - are still the same things covered? Even if the tests are equal, if less "setup" is needed (e.g., mocks), it is better.

I need to look into the code on my desktop machine to properly answer. Currently on the road.